File: p15663.java
import java.util.Arrays;
import java.util.Scanner;

public class p15663 {
	static int n, m, array[];
	static boolean visit[];

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);	
		n = in.nextInt();
		m = in.nextInt();
		array = new int[n];
		visit = new boolean[n];
		for(int i = 0; i < n; i++)
			array[i] = in.nextInt();
		Arrays.sort(array);

		dfs(0, "");
	}
	static int temp;
	public static void dfs(int len, String s) {
		if(len == m) {
			System.out.println(s);
			return;
		}
		else {
			for(int i = 0; i < n; i ++) {
				if(!visit[i]) {
					if(len == 0 || array[temp] != array[i]) {
						visit[i] = true;
						temp = i;
						dfs(len+1, s + array[i] + " ");
						visit[i] = false;
					}
				}


			}
		}
	}

}

File: p15663.cs
using System;
using System.Collections.Generic;
using System.Collections.Immutable;
using System.Text;

namespace bj
{
    class p15663
    {
        static SortedSet<string> set = new SortedSet<string>();
        static int n, m;
        static int[] array;
        static bool[] visit;
        static void Main()
        {
            String[] s = Console.ReadLine().Split(' ');
            n = int.Parse(s[0]);
            m = int.Parse(s[1]);
            array = new int[n];
            visit = new bool[n];
            s = Console.ReadLine().Split(' ');

            for (int i = 0; i < n; i++)
                array[i] = int.Parse(s[i]);

            Array.Sort(array);
            func(0, "");

            for (int i = 0; i < set.Count; i++)
                Console.WriteLine(set.ToImmutableArray()[i]);
        }

        static void func(int length, String s)
        {
            if (length == m)
            {
                set.Add(s.Trim());
                return;
            }
            else
            {
                for (int i = 0; i < n; i++)
                {
                    if (!visit[i])
                    {
                        visit[i] = true;
                        func(length + 1, s + array[i] + " ");
                        visit[i] = false;
                    }
                }
            }
        }
    }


}
